pkg/kubelet/kubelet_pods.go

```diff
@@ -326,7 +327,10 @@ func ensureHostsFile(fileName, hostIP, hostName, hostDomainName string, hostAlia
 		hostsFileContent = managedHostsFileContent(hostIP, hostName, hostDomainName, hostAliases)
 	}
 
-	return ioutil.WriteFile(fileName, hostsFileContent, 0644)
+	oldMask := unix.Umask(0)
```

Comment:
this change will not be portable on Windows as Umask() calls `RawSyscallNoError`.
if you still want to apply it, you'd need e.g. separate _unix.go, _windows.go variants of `kubelet_pods.go`.

Reply:
Thansk for a great advice :), changes have made.